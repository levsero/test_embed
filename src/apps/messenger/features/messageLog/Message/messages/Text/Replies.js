import React from 'react'
import PropTypes from 'prop-types'
import { getClient } from 'src/apps/messenger/suncoClient'

const Replies = ({ replies }) => {
  return (
    <div>
      {replies.map(({ payload, text }) => (
        <button
          onClick={() => {
            const client = getClient()
            client.sendMessage(payload)
          }}
        >
          {text}
        </button>
      ))}
    </div>
  )
}

Replies.propTypes = {
  replies: PropTypes.arrayOf(
    PropTypes.shape({
      role: PropTypes.string,
      text: PropTypes.string
    })
  )
}

export default Replies
