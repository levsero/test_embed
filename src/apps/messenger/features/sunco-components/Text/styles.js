import styled from 'styled-components'

const Text = styled.span`
  white-space: pre-wrap;
  word-wrap: break-word;
  max-width: ${props => props.theme.messenger.space.textMaxWidth};
  font-size: ${props => props.theme.messenger.fontSizes.md};
  line-height: ${props => props.theme.messenger.lineHeights.sm};
  width: auto;
`

const Padding = styled.div`
  padding-left: ${props => props.theme.messenger.space.sm};
  padding-right: ${props => props.theme.messenger.space.sm};
  padding-top: ${props => props.theme.messenger.space.xs};
  padding-bottom: ${props => props.theme.messenger.space.xs};
  width: 100%;
`

export { Text, Padding }
