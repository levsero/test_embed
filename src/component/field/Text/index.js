import React from 'react';
import PropTypes from 'prop-types';
import { renderLabel } from 'utility/fields';

import { TextField, Input, Hint, Message, Label } from '@zendeskgarden/react-textfields';

const Text = ({
  label,
  required,
  description,
  errorString,
  showError,
  inputProps
}) => {
  const validation = showError ? 'error' : 'none';

  return (
    <TextField>
      {renderLabel(Label, label, required)}
      {description && <Hint>{description}</Hint>}
      <Input {...inputProps} validation={validation} />
      {showError && <Message validation={validation}>{errorString}</Message>}
    </TextField>
  );
};

Text.propTypes = {
  label: PropTypes.string.isRequired,
  required: PropTypes.bool.isRequired,
  showError: PropTypes.bool.isRequired,
  description: PropTypes.string,
  errorString: PropTypes.string,
  inputProps: PropTypes.shape({
    type: PropTypes.string,
    pattern: PropTypes.object,
    value: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    required: PropTypes.bool,
  })
};

export default Text;
