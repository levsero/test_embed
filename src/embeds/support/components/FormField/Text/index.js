import React from 'react'
import PropTypes from 'prop-types'
import { Field, Hint, Input, Message } from '@zendeskgarden/react-forms'
import ContactFormLabel from 'src/embeds/support/components/FormField/ContactFormLabel'

const Text = ({ field, value, errorMessage, onChange }) => {
  return (
    <Field>
      {field.title_in_portal && (
        <ContactFormLabel value={field.title_in_portal} required={field.required_in_portal} />
      )}

      {field.description && <Hint>{field.description}</Hint>}

      <Input
        name={field.id}
        value={value || ''}
        required={Boolean(field.required_in_portal)}
        onChange={e => {
          onChange(e.target.value)
        }}
        validation={errorMessage ? 'error' : undefined}
      />
      {errorMessage && <Message validation="error">{errorMessage}</Message>}
    </Field>
  )
}

Text.propTypes = {
  field: PropTypes.shape({
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    title_in_portal: PropTypes.string,
    description: PropTypes.string,
    required_in_portal: PropTypes.bool
  }),
  value: PropTypes.string,
  onChange: PropTypes.func,
  errorMessage: PropTypes.string
}

export default Text
