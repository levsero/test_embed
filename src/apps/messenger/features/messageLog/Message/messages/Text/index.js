import React from 'react'
import PropTypes from 'prop-types'
import DummyStructuredMessage from '../DummyStructuredMessage'
import Replies from './Replies'

const extractReplies = message => {
  if (!message.isLastInLog) return null
  return message.actions?.filter(action => action.type === 'reply')
}

const TextMessage = ({ message }) => {
  const replies = extractReplies(message)
  return (
    <>
      <DummyStructuredMessage message={message} />
      {replies && <Replies replies={replies} />}
    </>
  )
}

TextMessage.propTypes = {
  message: PropTypes.shape({
    actions: PropTypes.array
  })
}

export default TextMessage
