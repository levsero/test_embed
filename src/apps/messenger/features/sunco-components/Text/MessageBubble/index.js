import React from 'react'
import PropTypes from 'prop-types'

import { PrimaryParticipantBubble, OtherParticipantBubble } from './styles'

const MessageBubble = ({ appUser, children, isFirstInGroup, isLastInGroup }) => {
  const ChosenBubble = appUser ? PrimaryParticipantBubble : OtherParticipantBubble

  const isStandalone = isFirstInGroup && isLastInGroup

  const first = !isStandalone && isFirstInGroup
  const middle = !isStandalone && !isFirstInGroup && !isLastInGroup
  const last = !isStandalone && isLastInGroup

  return (
    <ChosenBubble isStandalone={isStandalone} last={last} first={first} middle={middle}>
      {children}
    </ChosenBubble>
  )
}

MessageBubble.propTypes = {
  appUser: PropTypes.bool,
  children: PropTypes.node,
  isFirstInGroup: PropTypes.bool,
  isLastInGroup: PropTypes.bool
}

export default MessageBubble
