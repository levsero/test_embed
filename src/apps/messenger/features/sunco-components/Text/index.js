import React from 'react'
import PropTypes from 'prop-types'
import PrimaryParticipantLayout from 'src/apps/messenger/features/sunco-components/Layouts/PrimaryParticipantLayout'
import OtherParticipantLayout from 'src/apps/messenger/features/sunco-components/Layouts/OtherParticipantLayout'
import MessageBubble from 'src/apps/messenger/features/sunco-components/Text/MessageBubble'

import { Text, Padding } from './styles'

const SuncoTextMessage = ({ role, text, isFirstInGroup, isLastInGroup, avatar }) => {
  const appUser = role === 'appUser'

  const Layout = appUser ? PrimaryParticipantLayout : OtherParticipantLayout

  return (
    <Layout isFirstInGroup={isFirstInGroup} avatar={avatar}>
      <MessageBubble
        isFirstInGroup={isFirstInGroup}
        isLastInGroup={isLastInGroup}
        appUser={appUser}
      >
        <Padding>
          <Text>{text}</Text>
        </Padding>
      </MessageBubble>
    </Layout>
  )
}

SuncoTextMessage.propTypes = {
  role: PropTypes.string,
  text: PropTypes.string,
  avatar: PropTypes.string,
  isFirstInGroup: PropTypes.bool,
  isLastInGroup: PropTypes.bool
}

export default SuncoTextMessage
