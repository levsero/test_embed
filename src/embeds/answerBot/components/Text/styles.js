import styled from 'styled-components'

import { zdColorWhite, zdColorOil, zdColorAntiFlashWhite } from '@zendeskgarden/css-variables'
import { MessageBubble } from 'component/shared/MessageBubble'

const Container = styled.div`
  ${props => {
    return `
      margin-bottom: ${7 / props.theme.fontSize}rem;
      ${props.isVisitor ? userMessage(props) : botMessage(props)}
    `
  }}
`

const userMessage = props => `
  float: ${props.theme.rtl ? 'left' : 'right'};
  color: ${zdColorWhite};
`

const botMessage = props => `
  float: ${props.theme.rtl ? 'right' : 'left'};
  color: ${zdColorOil};
`

const Bubble = styled(MessageBubble)`
  ${props => {
    return `
      ${props.theme.rtl ? 'margin-right' : 'margin-left'}: ${40 / props.theme.fontSize}rem;
      ${props.isVisitor ? userBackground(props) : botBackground(props)}
    `
  }}
`

const userBackground = props => `
  background-color: ${props.theme.buttonColorStr} !important;
  color: ${props.theme.buttonTextColorStr} !important;

  &:hover,
  &:active,
  &:focus {
    background-color: ${props.theme.buttonHighlightColorStr} !important;
  }
`

const botBackground = props => `
  background: ${zdColorAntiFlashWhite};
  max-width: ${219 / props.theme.fontSize}rem !important;
`

export { Container, Bubble }
