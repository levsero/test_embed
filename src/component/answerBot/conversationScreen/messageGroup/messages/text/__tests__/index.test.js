import { render } from '@testing-library/react'
import React from 'react'

import Text from '../index'

test('renders the expected classes when not visitor', () => {
  const { container } = render(<Text isVisitor={false} message="not a visitor" />)

  expect(container).toMatchSnapshot()
})

test('renders the expected classes when visitor', () => {
  const { container } = render(<Text isVisitor={true} message="a visitor" />)

  expect(container).toMatchSnapshot()
})
