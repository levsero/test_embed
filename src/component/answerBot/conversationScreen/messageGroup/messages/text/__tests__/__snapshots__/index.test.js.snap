// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the expected classes when not visitor 1`] = `
<div>
  <div
    class="message botMessage"
  >
    <div>
      <div
        class="messageBubble messageBubble botBackground"
      >
        <span
          class="Linkify"
        >
          not a visitor
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the expected classes when visitor 1`] = `
<div>
  <div
    class="message userMessage"
  >
    <div>
      <div
        class="messageBubble messageBubble userBackground"
      >
        <span
          class="Linkify"
        >
          a visitor
        </span>
      </div>
    </div>
  </div>
</div>
`;
